### ISSUE TITLE
UX: Option to show original quality image when in full screen

### ISSUE THREAD

**Issue by ReneHollander** [photoprism/photoprism#3432]
**Describe what problem this solves and why this would be valuable to many users**

When looking at photos in the Web UI, even when full screen, a low quality converted JPEG is shown to the user. I think this detract from the viewing experience since it's often very obvious that the photo has been downscaled. It makes using the photo gallery for casual photo viewing not that great.

**Describe the solution you'd like**

An option to enable showing the original image in the Web UI when in full screen mode. If the file format is incompatible, a high quality on the fly converted version should be displayed instead.

**Describe alternatives you've considered**

The only option would be to increase the quality for converted images, which would greatly increase storage and lower usability when not needing the high quality images.

**Additional context**

n/a

**Comment by lastzero** [comment]
Already part of this issue:
- #1307

**Comment by ReneHollander** [comment]
Thanks, I assumed there was a FR about this somewhere that I just failed to find... :)

What are your thoughts on the suggestion that was also made by @CaptnSeraph:

> I wonder if a small interim fix would be a toggle for "full resolution" where instead of loading the fit_xxxx resolution it just forces the original to load within photoswipe?

I might spend a bit of time getting something working (even if it's just a environment variable to always enable this) in a PR.

**Comment by lastzero** [comment]
It's not worth making a interim fix. Right now, we are out of the office, so can't make any changes. Once we are back, we need to focus on one issue at a time without a lot of context switching or redoing the same work eg going from one workaround to the next.

In addition, the original image would leak metadata (thumbs do not) and might also show in the wrong oritentation (thumbs do not). Originals can also be serveral MB large, while thumbs generally are not. If you have a high res device, thumbs should already be loaded in the right resolution. Only exception is if the original has limited resolution or you have limited static thumb size while also having disabled dynamic thumb generation. To get the original, you can download it if needed eg to zoom in or use the original in other apps.

**Comment by lastzero** [comment]
Also note that you can already configure the thumbs API so that it returns the original image for high res screens. This was implemented long ago and is described in our docs.

**Comment by lastzero** [comment]
https://docs.photoprism.app/user-guide/settings/advanced/#dynamic-and-static-size-limits

**Comment by CaptnSeraph** [comment]
I agree that implementing this is not ideal as an interim measure at this point, However for those who use photoprism for themselves, revealing the metadata is not a problem, why would I care if my metadata is shared with me.

Makes me wonder if your planned use case is vastly different from my use case.

As to the oversized images option not having thumbnails, is it possible to configure a small enough maximum thumbnail size to essentially force original image loading through what might be considered a "misconfiguration" of the thumbnail sizes?

And if the original image is loaded, is it then stuck within some overly controlling image handling system?

The idea behind this full size image is to enable deep zoom (more than 100% zoom) which is a feature that is sorely lacking from photoprism if you wanted to be a lightroom cloud replacement (and frankly, quite a core feature for professional photographers)

**Comment by lastzero** [comment]
We know you might not care, but others do care and it could be unexpected. So ultimately such a workaround must be explicit, like when you follow our docs to force this as described, or extremely well documented so that everyone understands what is happening and what the consequences are. In that case, it's a lot of work and not a quick workaround anymore.

**Comment by CaptnSeraph** [comment]
Well firstly, any toggle could popup a "this option might expose your metadata to users, click here to acknowledge this risk" and that would solve that.

But it's ok, it's clear that your design and plans are for someone to share content with others, and that's not something I want to do at all, so it's ok for you to not cater to my needs as this project obviously has other use cases in mind.

On which note, I withdraw my feature request.

**Comment by lastzero** [comment]
Any message in the UI must be translated into more than 40 languages. Also having too many config options and the same functionality implemented in x different ways makes the application VERY hard to test for us and also VERY hard for the average user to understand.

**Comment by lastzero** [comment]
Let me say that we care for all users and use case. It's just that we right now need a short break after having released an extremely long list of enhancements and new features. As we have limited time, we then need to reduce context switching and the amount of technical debt. Most workarounds are technical debt and essentially mean having to implement the same functionality multiple times, which is something we simply cannot afford. That does NOT mean we don't care about you or other users.

**Comment by CaptnSeraph** [comment]
It's not that I think you don't care, rather, your priority list on your roadmap shows your priorities, and they don't align with mine.

Which is fine, and I'm neither upset nor concerned.

Not every tool is designed for every user, it can't be or there would be no reason to use one over another.

I migrated to my own implementation of something else about a year and a half after the proposed rewrite of the image viewer, which to my knowledge still hasn't been done, while a "memories of X date" is on priority which is a feature I hate in OneDrive and Google photos.

But only because I use it for professional photo management and you are aimed at cloud hosting replacement.

At the time I thought there might have been enough overlap that I could use photoprism, but no longer.

And that's not a bad mark against you or your team, it's simply a divergence of interest and use cases.

Your attempt to chase my use would as you rightly say, cause additional workload, add menu options that would confuse ordinary users, and in some cases directly go against your main use case.

Don't chase the edge cases.

Also, if you need translation work, let me know, I can translate to about 50 or so languages with a fairly high degree of accuracy with AI.

**Comment by atahrijouti** [comment]
Hi @lastzero, are there any plans for making this option happen? if not, what's the next best thing that doesn't involve making the previews bigger? which would take much more space on disk, rather than just showing the original when clicking on the photo right there and then?

This might not be the way you're deciding to move forwards with, but I thought that
- showing small previews in the gallery, helps having fluid browsing
- opening the photo to see it better, might include the option to view the full quality image if possible.
  - this keeps the thumbnails still small
  - gets the user the full quality image when they want to

**Comment by lastzero** [comment]
It already is possible to enlarge (zoom into) images and live photos:
- https://github.com/photoprism/photoprism/issues/4704

**Comment by atahrijouti** [comment]
Ah thanks
